kernel/src/sched/sched_class/mod.rs

```diff
@@ -214,8 +217,27 @@ impl Scheduler for ClassScheduler {
         }
 
         thread.sched_attr().set_last_cpu(cpu);
-        rq.enqueue_entity((task, thread), Some(flags));
-        Some(cpu)
+        rq.enqueue_entity((task, thread.clone()), Some(flags));
```

Comment:
Could we compute `should_preempt` before this line to avoid the unnecessary `clone`?

```rust
        let should_preempt = { .. };

        thread.sched_attr().set_last_cpu(cpu);
        rq.enqueue_entity((task, thread), Some(flags));

        if should_preempt { .. } else { .. }
```